src/ApiManagement/ApiManagement.ServiceManagement.Test/ScenarioTests/ApiManagementTests.ps1

```diff
@@ -3618,4 +3618,161 @@ function ApiDiagnostic-CrudTest {
+        Assert-AreEqual $locationDistrict $gateway.LocationData.District
+        Assert-AreEqual $locationRegion $gateway.LocationData.CountryOrRegion
+
+        #get
+        $gateway = Get-AzApiManagementGateway -Context $context -GatewayId $gatewayId
+        Assert-NotNull $gateway
+        Assert-AreEqual $description $gateway.Description
+        Assert-AreEqual $locationCity $gateway.LocationData.City
+        Assert-AreEqual $locationName $gateway.LocationData.Name
+        Assert-AreEqual $locationDistrict $gateway.LocationData.District
+        Assert-AreEqual $locationRegion $gateway.LocationData.CountryOrRegion
+
+        #update
+        $newDescription = getAssetName
+        $gateway = Update-AzApiManagementGateway -Context $context -GatewayId $gatewayId -Description $newDescription -PassThru
+        Assert-NotNull $gateway
+        Assert-AreEqual $newDescription $gateway.Description
+        Assert-AreEqual $locationCity $gateway.LocationData.City
+        Assert-AreEqual $locationName $gateway.LocationData.Name
+        Assert-AreEqual $locationDistrict $gateway.LocationData.District
+        Assert-AreEqual $locationRegion $gateway.LocationData.CountryOrRegion
+
+        #update location
+        $newLocationCity = getAssetName
+        $location = New-AzApiManagementResourceLocation -Name $locationName -City $newLocationCity
+        $gateway = Update-AzApiManagementGateway -Context $context -GatewayId $gatewayId -LocationData @location -PassThru
+        Assert-NotNull $gateway
+        Assert-AreEqual $newDescription $gateway.Description
+        Assert-AreEqual $newLocationCity $gateway.LocationData.City
+        Assert-AreEqual $locationName $gateway.LocationData.Name
+        Assert-Null $gateway.LocationData.District
+        Assert-Null $gateway.LocationData.CountryOrRegion
+
+        #get keys
+        $key = Get-AzApiManagementGatewayKey -Context $context -GatewayId $gatewayId
+        Assert-NotNull $key
+        Assert-NotNull $key.PrimaryKey
+        Assert-NotNull $key.SecondaryKey
+
+        #attach to api
+        $apis = Get-AzApiManagementApi -Context $context
+        $apiId = $apis[0].ApiId
+
+        Add-AzApiManagementApiToGateway -Context $context -GatewayId $gatewayId -ApiId $apiId  
+
+        $apis = Get-AzApiManagementApi -Context $context -GatewayId $gatewayId
+        Assert-AreEqual 1 $apis.Count
+
+        #detach from api
+        Remove-AzApiManagementApiFromGateway -Context $context -GatewayId $gatewayId -ApiId $apiId -PassThru
+
+        $apis = Get-AzApiManagementApi -Context $context -GatewayId $gatewayId
+        Assert-AreEqual 0 $apis.Count
+
+        $certPath = Join-Path (Join-Path "$TestOutputRoot" "Resources") "powershelltest.pfx"
+        #[SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine")]
+        $certPassword = 'Password'
+        $certSubject = "CN=*.msitesting.net"
+        $certThumbprint = '8E989652CABCF585ACBFCB9C2C91F1D174FDB3A2'
```

Comment:
>$certThumbprint = '8E989652CABCF585ACBFCB9C2C91F1D174FDB3A2' [](start = 8, length = 60)

you are not actually this property `certThumbprint`

Reply:
Removed